drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -369,6 +349,23 @@ def full_name
     [first_name, middle_name, last_name, name_suffix].compact.join(' ')
   end
 
+  def build_client_headshot_file(image_blob_id, current_user)
+    current_image_blob = ActiveStorage::Blob.find_by(id: image_blob_id)
+    return unless current_image_blob
+
+    # Note: this builds a GrdaWarehouse::ClientFile where the associated client is a Hmis::Hud::Client,
+    # (not a GrdaWarehouse::Hud::Client), which may lead to unexpected behavior. Skip callbacks to avoid issues.
+    file = client_files.build(
+      user_id: current_user.id,
+      name: 'Client Headshot',
+      visible_in_window: false,
+    )
+    file.callbacks_skipped = true
+    file.tag_list.add('Client Headshot') # Special string in ClientFileBase, don't change
```

Comment:
Let's make this a class method in `ClientFileBase`
```
def self.headshot_tag_name
  'Client Headshot'
end
```

then this become:
```suggestion
    file.tag_list.add(ClientFileBase.headshot_tag_name)
```

That should keep us in sync in the future.

Reply:
I think the suggestion is slightly off since `ClientFileBase` is a module

Reply:
Thanks yeah this became `file.tag_list.add(GrdaWarehouse::ClientFile.headshot_tag_name)` on https://github.com/greenriver/hmis-warehouse/pull/4837/commits/127b8adf6b76cea328c4fa8976dc042ab7986e39